src/services/lesson/hooks/index.js

```diff
@@ -59,6 +61,55 @@ const mapUsers = (context) => {
 	return context;
 };
 
+const validateLessonFind = (context) => {
+	const query = (context.params || {}).query || {};
+	if (query.courseId || query.courseGroupId || query.shareToken) {
+		return context;
+	}
+	throw new BadRequest('this operation requires courseId, courseGroupId, or shareToken');
+};
+
+const getCourseAndCourseGroup = async (courseId, courseGroupId, app) => {
+	let tempCourseId = courseId;
+	let courseGroup;
+	if (courseGroupId) {
+		courseGroup = await app.service('courseGroupModel').get(courseGroupId);
+		tempCourseId = courseGroup.courseId;
+	}
+	const course = await app.service('courseModel').get(tempCourseId);
+	return { course, courseGroup };
+};
+
+const restrictToUsersCoursesLessons = async (context) => {
+	const { userId, schoolId } = context.params.account;
+	const userIsSuperhero = await hasRoleNoHook(context, userId, 'superhero');
+	const userIsAdmin = await hasRoleNoHook(context, userId, 'administrator');
+
+	let courseId;
+	let courseGroupId;
+	if (context.method === 'find') {
+		if (context.params.query.shareToken) return context;
+		({ courseId, courseGroupId } = context.params.query);
+	} else {
+		const lesson = await context.app.service('lessons').get(context.id);
+		({ courseId, courseGroupId } = lesson);
+	}
+
+	const { course, courseGroup } = await getCourseAndCourseGroup(courseId, courseGroupId, context.app);
+	let studentsWithAccess = course.userIds;
+	if (courseGroup) studentsWithAccess = courseGroup.userIds;
+
+	const hasAdminAccess = userIsSuperhero || (userIsAdmin && equal(course.schoolId, schoolId));
+	const userInCourse = studentsWithAccess.some((id) => equal(id, userId))
+		|| course.teacherIds.some((id) => equal(id, userId))
+		|| course.substitutionIds.some((id) => equal(id, userId));
```

Comment:
Why ubstitutionIds?

Reply:
because substitution teachers need to access the lessons.